### PR TITLE
Fix Issue 17956 - core.memory unittest failure (possibly glibc 2.26 s…

### PR DESCRIPTION
**Pull request by wilzbach**
…pecific)

Let's see what auto-tester thinks about this.

### CODE REVIEW

**Comment by ibuclaw** [comment]
It is glibc 2.26 specific (there is a CVE for it), and it should be applied to a latter 2.26 point release.

File: src/core/memory.d [hunk 1]
```diff
@@ -893,7 +893,7 @@ void pureFree(void* ptr) @system pure @nogc nothrow
 
     // subtract 2 because snn.lib adds 2 unconditionally before passing
     //  the size to the Windows API
-    void* z = pureMalloc(size_t.max - 2); // won't affect `errno`
+    void* z = pureMalloc(long.max - 2); // won't affect `errno`
```
Comment (ibuclaw) [inline src/core/memory.d]: This probably would be ptrdiff_t for the benefit of 32bit systems? I don't recall the signature of gc.malloc within druntime.

Comment (ibuclaw) [reply src/core/memory.d]: In any case, the bug applies only when you call malloc with a size from `size_t.max-23ʼ and up.

Shifting the value by one bit would be sufficient also (`size_t.max >> 1`)

Comment (ibuclaw) [reply src/core/memory.d]: Or unsetting the right most 8 bits to keep the adjustment small (`size_t.max & ~255`).

Its all just a workaround of a libc bug anyway. ☺️

Comment (wilzbach) [reply src/core/memory.d]: > Or unsetting the right most 8 bits to keep the adjustment small (size_t.max & ~255).

Seems like that worked fine :)

**Pull request merged**